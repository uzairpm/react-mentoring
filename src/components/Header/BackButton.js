import React, {Component} from 'react';
import { Link } from "react-router-dom";

const BackButton = (props) => (
    <Link to="/" className="btn btn-primary btn-sm widerButton">SEARCH</Link>
);
export default BackButton;
