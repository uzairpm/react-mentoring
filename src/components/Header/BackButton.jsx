import React from 'react';
import { Link } from "react-router-dom";
import styled from 'styled-components';

const WiderLink = styled(Link)`
    padding: 0.25rem 2rem;
`;

const BackButton = (props) => (
    <WiderLink to="/" className="btn btn-primary btn-sm">SEARCH</WiderLink>
);
export default BackButton;
