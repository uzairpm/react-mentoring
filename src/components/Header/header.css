.navbar-light .navbar-brand.navbarItem {
    font-size: 1.4rem;
    font-Weight: 500;
    color: #fa6c85;
};
.widerButton {
    padding: 0.25rem 2rem !important;
}