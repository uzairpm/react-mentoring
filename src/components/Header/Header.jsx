import React from 'react';
import styled from 'styled-components';

import BackButton from './BackButton';
import SearchBoxContainer from '../Search/containers/SearchBoxContainer';

const BrandLink = styled.a`
    font-size: 1.4rem;
    font-Weight: 500;
    color: #fa6c85 !important;
`;

const Header = (props) => (
    <div className = "container">
        <nav className="navbar navbar-light">
            <BrandLink className="navbar-brand" href="#">netflixroulette</BrandLink>
            { props.showSearchButton && <BackButton /> }
        </nav>
        { props.showSearchButton === false && <SearchBoxContainer /> }
    </div>
);
export default Header;
