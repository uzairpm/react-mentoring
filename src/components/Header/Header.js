import React, {Component} from 'react';
import { Link } from "react-router-dom";

import Button from '../Common/Button';
import './header.css';

export default class Header extends Component {
    constructor(props) {
        super(props);
        this.searchClick = this.searchClick.bind(this);
    }
    searchClick(ev) {
        ev.preventDefault();
    }

    render() {
        var view = '';
        if (this.props.showSearchButton === true) {
            view = (
                <form className="form-inline">
                    <Link to="/" className="btn btn-primary btn-sm widerButton">SEARCH</Link>
                </form>
            );
        }
        
        return (
            <div className = "container">
                <nav className="navbar navbar-light">
                    <a className="navbar-brand navbarItem" href="#">netflixroulette</a>
                    {view}
                </nav>
            </div>
        );
    }
}
