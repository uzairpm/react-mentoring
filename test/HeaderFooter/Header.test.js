import { shallow } from 'enzyme';
import React from 'react';

import Header from '../../src/components/Header/Header';

describe('Header Component', () => {
    test('Snapshot test when search button is hidden', () => {
        const element = shallow(<Header showSearchButton={false}/>);
        expect(element).toMatchSnapshot();
    });
    test('Snapshot test when search button is displayed', () => {
        const element2 = shallow(<Header showSearchButton={true}/>);
        expect(element2).toMatchSnapshot();
    });
});
