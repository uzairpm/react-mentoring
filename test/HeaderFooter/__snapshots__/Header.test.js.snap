// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot test when search button is displayed 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    showSearchButton={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="navbar navbar-light"
        >
          <a
            className="navbar-brand navbarItem"
            href="#"
          >
            netflixroulette
          </a>
          <BackButton />
        </nav>,
        false,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand navbarItem"
              href="#"
            >
              netflixroulette
            </a>,
            <BackButton />,
          ],
          "className": "navbar navbar-light",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "netflixroulette",
              "className": "navbar-brand navbarItem",
              "href": "#",
            },
            "ref": null,
            "rendered": "netflixroulette",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "nav",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="navbar navbar-light"
          >
            <a
              className="navbar-brand navbarItem"
              href="#"
            >
              netflixroulette
            </a>
            <BackButton />
          </nav>,
          false,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="navbar-brand navbarItem"
                href="#"
              >
                netflixroulette
              </a>,
              <BackButton />,
            ],
            "className": "navbar navbar-light",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "netflixroulette",
                "className": "navbar-brand navbarItem",
                "href": "#",
              },
              "ref": null,
              "rendered": "netflixroulette",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "nav",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Header Component Snapshot test when search button is hidden 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    showSearchButton={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <nav
          className="navbar navbar-light"
        >
          <a
            className="navbar-brand navbarItem"
            href="#"
          >
            netflixroulette
          </a>
        </nav>,
        <SearchBox
          genreClick={undefined}
          searchClick={undefined}
          titleActive={undefined}
          titleClick={undefined}
          valChange={undefined}
          value={undefined}
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="navbar-brand navbarItem"
              href="#"
            >
              netflixroulette
            </a>,
            false,
          ],
          "className": "navbar navbar-light",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "netflixroulette",
              "className": "navbar-brand navbarItem",
              "href": "#",
            },
            "ref": null,
            "rendered": "netflixroulette",
            "type": "a",
          },
          false,
        ],
        "type": "nav",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "genreClick": undefined,
          "searchClick": undefined,
          "titleActive": undefined,
          "titleClick": undefined,
          "valChange": undefined,
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <nav
            className="navbar navbar-light"
          >
            <a
              className="navbar-brand navbarItem"
              href="#"
            >
              netflixroulette
            </a>
          </nav>,
          <SearchBox
            genreClick={undefined}
            searchClick={undefined}
            titleActive={undefined}
            titleClick={undefined}
            valChange={undefined}
            value={undefined}
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="navbar-brand navbarItem"
                href="#"
              >
                netflixroulette
              </a>,
              false,
            ],
            "className": "navbar navbar-light",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "netflixroulette",
                "className": "navbar-brand navbarItem",
                "href": "#",
              },
              "ref": null,
              "rendered": "netflixroulette",
              "type": "a",
            },
            false,
          ],
          "type": "nav",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "genreClick": undefined,
            "searchClick": undefined,
            "titleActive": undefined,
            "titleClick": undefined,
            "valChange": undefined,
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
